File: core/src/main/java/com/fahamutech/kcore/admin/database/noSql/NoSqlDatabase.java
package com.fahamutech.kcore.admin.database.noSql;

import android.content.Context;

import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.FirebaseFirestoreSettings;

public abstract class NoSqlDatabase {

    FirebaseFirestore firestore;
    protected Context context;

    NoSqlDatabase(Context context) {
        FirebaseFirestoreSettings settings = new FirebaseFirestoreSettings.Builder()
                .setPersistenceEnabled(true)
                .setSslEnabled(true)
                .build();
        this.firestore = FirebaseFirestore.getInstance();
        this.firestore.setFirestoreSettings(settings);
        this.context = context;
    }


}

File: core/src/main/java/com/fahamutech/kcore/admin/database/noSql/NoSqlDatabase.kt
package com.fahamutech.kcore.admin.database.noSql

import android.content.Context
import com.fahamutech.kcore.utils.getKCoreFirebaseApp
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.FirebaseFirestoreSettings

abstract class NoSqlDatabase internal constructor(context: Context) {
    @JvmField
    var firestore: FirebaseFirestore
    @JvmField
    protected var context: Context

    init {
        val settings = FirebaseFirestoreSettings.Builder()
            .setPersistenceEnabled(true)
            .setSslEnabled(true)
            .build()
        firestore = FirebaseFirestore.getInstance(getKCoreFirebaseApp(context))
        firestore.firestoreSettings = settings
        this.context = context
    }
}